Enhance/display important dates on search

millerf opened the pull request:
#### Changes proposed in this Pull Request

* Depending on post/page status, display the most meaningful date
* On search, display the said date with the status label
* ~Small~ Good Refactor of `PostRelativeTime`

 Search results should look close to this mockup:
![Screenshot 2020-04-15 at 00 31 49](https://user-images.githubusercontent.com/790558/79280455-b1c45780-7eb0-11ea-979a-84a6532dfa36.png)




#### Testing instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Go to pages or posts views. Make sure displayed dates are the right ones
* When searching, make sure the dates are the same and label is displayed



#### Extras
Found a bug reported here
https://github.com/Automattic/wp-calypso/issues/41245
As it is out of scope for this one.


Fixes #41075

cc @mmtr @sixhours

millerf committed:
Display the right date all the time

millerf committed:
Display the right date all the time

millerf committed:
Small refactor as post cannot be null (props.isRequired)

millerf committed:
Small refactor as post cannot be null (props.isRequired)

millerf committed:
clean up - includeNonDraftStatuses was never used across the project

millerf committed:
clean up - includeNonDraftStatuses was never used across the project

millerf committed:
Fix CamelCase

millerf committed:
Fix CamelCase

millerf committed:
More refactor.
Added "sticky" status for posts on search and listing

millerf committed:
More refactor.
Added "sticky" status for posts on search and listing

millerf committed:
Refactor label, use only one method to display time label
Add sticky Label for posts

millerf committed:
Refactor label, use only one method to display time label
Add sticky Label for posts

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,141 +20,158 @@ class PostRelativeTime extends React.PureComponent {
-		return time.fromNow();
+		// If the content is scheduled to be release within a year, do not display the year at the end
+		const scheduledTime = scheduledDate.calendar( null, {
+			sameElse: this.props.translate( '[on] ll [at] LT', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **20** times:
`translate( 'LL [at] LT' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/pull/41242)

millerf replied:
`'LL [at] LT' ` has been modified and is now ` '[on] ll [at] LT'`

a8ci18n commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,141 +20,158 @@ class PostRelativeTime extends React.PureComponent {
-				statusText = this.props.translate( 'Publish immediately' );
-			}
+			const displayScheduleTime = this.getDisplayedTimeForLabel();
+			statusText = this.props.translate( 'scheduled %(displayScheduleTime)s', {
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **3** times:
`translate( 'scheduled for %(displayScheduleTime)s' )` *ES Score: 8*

millerf replied:
`'scheduled for %(displayScheduleTime)s'` has been modified and is now  `'scheduled %(displayScheduleTime)s'`

millerf commented:
`PostRelativeTime` was refactored.

Posts and Pages views were modified according to specs:


| Before | After | Comment|
| ---- | ---- | ----|
| ![Screenshot 2020-04-18 at 19 21 32](https://user-images.githubusercontent.com/790558/79644548-f3534c00-81a9-11ea-8f0b-182e5f3bfbe5.png) | ![Screenshot 2020-04-18 at 19 21 17](https://user-images.githubusercontent.com/790558/79644545-edf60180-81a9-11ea-9b9b-7be8ac429ee4.png) |  On listing, scheduled posts/pages now display the right date (scheduled ate) |
| ![Screenshot 2020-04-18 at 19 23 56](https://user-images.githubusercontent.com/790558/79644585-331a3380-81aa-11ea-8a6f-1e67a280ff57.png) |  ![Screenshot 2020-04-18 at 19 25 06](https://user-images.githubusercontent.com/790558/79644613-60ff7800-81aa-11ea-8b40-186fdbed6450.png) | Published now display "sticky" if needed | 
| ![Screenshot 2020-04-18 at 19 29 19](https://user-images.githubusercontent.com/790558/79644715-fef34280-81aa-11ea-9a0c-a3cd2b99fb72.png)  |  ![Screenshot 2020-04-18 at 19 26 25](https://user-images.githubusercontent.com/790558/79644651-91471680-81aa-11ea-97f4-afd2a801a14c.png)  | Post Search now displays the right status and appropriate date. The "sticky" status has been added. Maybe need some feedback from design? |
|  ![Screenshot 2020-04-18 at 19 29 05](https://user-images.githubusercontent.com/790558/79644751-45e13800-81ab-11ea-8dec-70fcf8510f54.png) | ![Screenshot 2020-04-18 at 19 31 03](https://user-images.githubusercontent.com/790558/79644737-2fd37780-81ab-11ea-813f-d060cce6b007.png)| Pages search now displays only one date and the right status |

ramonjd committed:
Gutenboarding: add start and complete tracking events for the main flow (#41129)

* initial commit:
- adding methods for start and end
- adding a constant for the flow name (could also be used for API calls, e.g., sites/new)

* refactored the start and complete to use a method with a default flow id
Changed the `gutenboarding` signup start track id with `calypso_signup_step_enter`

* Adding calypso_signup_complete tracking with new site and new user props. The current signup framework also passes `hasCartItems`, but I think we'll need a way to store that before we can pass it given that we do a redirect to the checkout

* Fix linting issue, add newUser to Header useEffect dependency array

* Update client/landing/gutenboarding/utils/analytics.ts

Correct jsdoc usage  comment

Co-Authored-By: roo2 <roo2@users.noreply.github.com>

* Remove useless comment.

* Moved analytics from /utils to /lib/analytics as utils seem to contain pure functions/no side effects. We can think about where to put analytics later, and whether to add to a store for easier testing similar to the Calypso analytics actions

Co-authored-by: Andrew Serong <andrewserong@users.noreply.github.com>
Co-authored-by: roo2 <roo2@users.noreply.github.com>

p-jackson committed:
Project: Remove arg forwarding "--" from yarn commands (#41263)

Aurorum committed:
Customer Home: Updating Timing of Connect Accounts Task (#41271)

daledupreez committed:
Add suffix to railcar identifier (#41197)

* Add suffix to railcar identifier
* Shift the railcar identifier logic up one level

adamziel committed:
Enable private atomic sites for 5% of new users (#40846)

enejb committed:
Jetpack: Add scan to the connect page (#41180)

Commit:
Include the site id in `calypso_signup_complete` event. (#41264)

* Include the site id in `calypso_signup_complete` event.

Per request by p2-pau2Xa-12j, this commit adds the site id to
`calypso_signup_complete` event. In the same vein, `isNewSite` becomes a
derived data within the function itself, hence removing it from the call
sites.

* Update `site_id` as `blog_id` since the latter is what Tracks expects.

sgomes committed:
Modularise login state (#40610)

* Modularise login state

* Create directory for login actions

* Split out remoteLoginUser

* Split out loginUser

* Split out updateNonce

* Split out loginUserWithSecurityKey

* Split out loginUserWithTwoFactorVerificationCode

* Split out loginSocialUser

* Fix bad imports

* Split out createSocialUser

* Split out connectSocialUser

* Split out disconnectSocialUser

* Split out createSocialUserFailed

* Split out sendSmsCode

* Split out push-related actions

* Split out account type-related actions

* Split out formUpdate

* Sort exports alphabetically

* Reorganise 2fa push poll code

* Remove unnecessary auth-options inits

* Fix addReducer on login entry point

* Extract addReducerEnhancer into a util

sgomes committed:
Async load 2FA login code (#41223)

eltongo committed:
Domains: Extend primary domain restriction to mappings (#41139)

hambai committed:
Show the destination in the domains navigation item for name servers and DNS (#41019)

Show the destination of a domain - does it point to WordPress.com site or is external - in the "Change your name servers & DNS records" nav item

eltongo committed:
Remove old domain status components (#41196)

scinos committed:
Upgrade to wp-prettier 2.0.4

scinos committed:
Automatic reformat with `yarn reformat-files`

sirreal committed:
Data-stores: Provide domain suggestions vendor (#41177)

* Rework Selectors as class to manage `vendor`

* Provide suggestions vendor to store registration

enejb committed:
Jetpack Cloud: Scanner display threats (#41216)

* Jetpack Cloud: Scanner display threats

* Fixes issues with typing.

* Adds threat type parser.

* Cleans up unused code and adds handling for idle state.

* Moves some logic to description component.

* Fixes style bugs with spacing.

* Adds typing for threat fixing.

* Adds logic around fixers and additional typing.

* Fixes position of fixing buttons on mobile.

* Fix buttons layout on mobile

Co-authored-by: Echo G <ChaosExAnima@users.noreply.github.com>
Co-authored-by: Renzo Canepa <rcanepag@gmail.com>

Commit:
Add Earth Day Live setting (#41275)

* Add Earth Day Live setting toggle to settings

* Add moment import

* Update copy as per editorial feedback

sixhours commented:
Thanks @millerf ! It looks like this needs to be rebased, it's asking me to use `npm` when we recently switched to `yarn`. Give that a try and let us know if you need help with it!

millerf committed:
Merge branch 'enhance/display_important_dates_on_search' of https://github.com/millerf/wp-calypso into enhance/display_important_dates_on_search

sixhours reviewed (commented):
This is not a comprehensive review. Please disregard if this is still "in progress"!

I noticed a couple issues when I fired this up:

When searching posts, the date shows "Published modified on [date]":
<img width="404" alt="Screen Shot 2020-04-20 at 11 48 25 AM" src="https://user-images.githubusercontent.com/2124984/79775686-9b7c2880-8302-11ea-8967-61d2df49ed29.png">

When searching for a scheduled post, the line "Scheduled for Tomorrow at [time]" reads oddly because "tomorrow" is capitalized:
<img width="357" alt="Screen Shot 2020-04-20 at 12 25 31 PM" src="https://user-images.githubusercontent.com/2124984/79775604-80111d80-8302-11ea-8b1f-d7f6188b0a1c.png">

The latter issue isn't unique to this PR (I noticed the same behavior on prod) but might be something we could address. Totally OK if that's out of scope and we address in a different PR: let me know and I'll create an issue.

millerf commented on client/my-sites/pages/page-card-info/index.jsx:
```diff
@@ -62,7 +62,7 @@ function PageCardInfo( {
 						link={ contentLink.contentLinkURL }
 						target={ contentLink.contentLinkTarget }
-						gridiconSize={ ICON_SIZE }
+						gridIconSize={ ICON_SIZE }
```
I'll revert the commit

millerf commented on client/blocks/post-item/index.jsx:
```diff
@@ -211,7 +211,7 @@ class PostItem extends React.Component {
 										link={ enabledPostLink }
 										target={ null }
-										gridiconSize={ ICON_SIZE }
+										gridIconSize={ ICON_SIZE }
```
Damn... I tried to look smart... 😄

millerf commented:
> "Published modified on [date]"

I see. What would be the correct way to say it... the idea is that the status of the post is published but the date is the last modified date.
Would "Published, last modified on [date]" be acceptable?

> "Scheduled for Tomorrow at [time]" 

It is definitely in the scope. I'll fix it.

I am having some trouble to push since I rebased. I don't think it is linked to the rebase, as I seem to have the same trouble on new branches...
```
refusing to allow an OAuth App to create or update workflow `.github/workflows/full-site-editing-plugin.yml` without `workflow` scope
``` 
So not sure if I would be able to address all that tonight.

sixhours commented:
> Would "Published, last modified on [date]" be acceptable?

Yes, that sounds right to me!

> It is definitely in the scope. I'll fix it.

Yay! Thank you!

> I am having some trouble to push since I rebased. I don't think it is linked to the rebase, as I seem to have the same trouble on new branches...

That's a new one to me. 😬 It sounds like something weird with Git/Github? Hoping @kwight can offer some insight.

millerf commented:
> It sounds like something weird with Git/Github? Hoping @kwight can offer some insight.

It is bizarre as I am pushing on my fork... Also using command-line. All help I found online relate to git GUIs... I might just scratch the local repo...

sixhours commented:
> I might just scratch the local repo...

That's my go-to for fixing weirdness, too. Burn it all to the ground. :D

millerf commented:
> > I might just scratch the local repo...
> 
> That's my go-to for fixing weirdness, too. Burn it all to the ground. :D

Wow... I deleted the local repo, updated git, nothing worked. I ended up downloading Github Desktop and it worked straight away... Something about OAuth tokens that weren't saved the right way or something. Github desktop manages it in the automatically in the background and it worked...

millerf commented on client/my-sites/post-relative-time-status/index.jsx:
```diff
@@ -21,141 +20,158 @@ class PostRelativeTime extends React.PureComponent {
 		link: PropTypes.string,
 		target: PropTypes.string,
-		gridiconSize: PropTypes.number,
+		gridIconSize: PropTypes.number,
```
resolved on #41301

millerf commented:
Work continues on #41301
